src/Mvc/Mvc.Core/test/Formatters/TranscodingReadStreamTest.cs

```diff
@@ -88,27 +88,31 @@ public async Task ReadAsync_WithOverflowBuffer()
         {
             // Arrange
             // Test ensures that the overflow buffer works correctly
-            var input = new string('A', 4096 + 4);
+            var input = "☀";
```

Comment:
Does it make sense to do some tests where this appears on different boundaries?

Reply:
It's a bit tricky to author tests that exactly validate this behavior, but I added a couple of tests that attempt to do some boundary value testing in an update